api/loan-position.ts

```diff
@@ -0,0 +1,36 @@
+import { LoanPosition } from '../model/loan-position'
+import { getBuiltGraphSDK } from '../.graphclient'
+
+import { toCurrency } from './asset'
+
+const { getLoanPositions } = getBuiltGraphSDK()
+
+export async function fetchLoanPositions(
+  userAddress: `0x${string}`,
+): Promise<LoanPosition[]> {
+  const { loanPositions } = await getLoanPositions(
+    {
+      userAddress: userAddress.toLowerCase(),
+    },
+    {
+      url: process.env.SUBGRAPH_URL,
+    },
+  )
+  return loanPositions.map((loanPosition) => ({
+    positionId: BigInt(loanPosition.id),
+    substitute: toCurrency(loanPosition.substitute),
+    underlying: toCurrency(loanPosition.underlying),
+    collateral: {
+      underlying: toCurrency(loanPosition.collateral.underlying),
+      substitute: toCurrency(loanPosition.collateral.substitute),
+      liquidationThreshold: loanPosition.collateral.liquidationThreshold,
```

Comment:
I think BigInt is appropriate here, even though the number is always safe. Numbers that use decimals precision should be `bigint`

Reply:
@onemonster, no, it can't be because Asset is used in getServerSideProps.

Reply:
Agreed but doesn't `toCurrency` make use of `bigint` as well? Just wondering why currency works and bigint doesn't.